### PR TITLE
Add remove entity node

### PR DESCRIPTION
**Pull request by Wiktor-99**
# 🎉 New feature

## Summary
Add remove entity node (complementary to create) 

## Test it
Run gz sim with any entity and call remove launch as follows:
```bash
ros2 launch ros_gz_sim gz_remove_model.launch.py world:='world_name' entity_to_remove_name:='entity_name'
```
Test are also provided.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

### CODE REVIEW

File: ros_gz_sim/launch/gz_remove_model.launch.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+def generate_launch_description():
+
+    world = LaunchConfiguration('world')
+    entity_to_remove_name = LaunchConfiguration('entity_to_remove_name')
+
+    declare_world_cmd = DeclareLaunchArgument(
+        'world', default_value=TextSubstitution(text=''),
+        description='World name')
+    declare_entity_to_remove_name_cmd = DeclareLaunchArgument(
```
Comment (caguero) [inline ros_gz_sim/launch/gz_remove_model.launch.py]: For consistency with `gz_spawn_model.launch.py`, I suggest to rename `entity_to_remove_name` to `entity_name` and propagate this change to any other place where this parameter was used.

Comment (Wiktor-99) [reply ros_gz_sim/launch/gz_remove_model.launch.py]: Done

File: ros_gz_sim/CMakeLists.txt [hunk 2]
```diff
@@ -48,7 +48,17 @@ target_link_libraries(create
   gz-msgs::core
   gz-transport::core
 )
-ament_target_dependencies(create std_msgs)
```
Comment (ahcorde) [inline ros_gz_sim/CMakeLists.txt:51]: restore, [create](https://github.com/gazebosim/ros_gz/blob/ros2/ros_gz_sim/src/create.cpp#L34) is using `std_msgs`

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: ros_gz_sim/CMakeLists.txt (hunk 2 above)
Comment (Wiktor-99) [reply ros_gz_sim/CMakeLists.txt:51]: Check line  43 std_msgs is already linked to create

File: ros_gz_sim/CMakeLists.txt [hunk 3]
```diff
@@ -104,13 +114,20 @@ install(FILES
...
 install(TARGETS
   create
   DESTINATION lib/${PROJECT_NAME}
 )
+
+install(TARGETS
+  remove
+  DESTINATION lib/${PROJECT_NAME}
+)
+
```
Comment (ahcorde) [inline ros_gz_sim/CMakeLists.txt]:
```suggestion
install(TARGETS
  create
  remove
  DESTINATION lib/${PROJECT_NAME}
)
```

File: ros_gz_sim/src/remove.cpp [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gz/msgs/boolean.pb.h>
+#include <gz/msgs/entity.pb.h>
+#include <gz/msgs/stringmsg_v.pb.h>
+
+#include <string>
```
Comment (ahcorde) [inline ros_gz_sim/src/remove.cpp:19]: include `<memory>`

**Review by ahcorde — CHANGES_REQUESTED** [review]
just small nits, otherwise LGTM

**Comment by Wiktor-99** [comment]
Done

**Review by ahcorde — APPROVED** [review]

**Comment by ahcorde** [comment]
https://github.com/Mergifyio backport jazzy

**Pull request merged**